streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTest.java

```diff
@@ -0,0 +1,962 @@
+                if (comment != null) {
+                    assertThat(
+                        pass,
+                        report,
+                        "comment articleId mismatch",
+                        comment.getArticleId(),
+                        is(augmentedComment.getArticleId())
+                    );
+                }
+                commentCounts.put(augmentedComment.getArticleId(),
+                                  commentCounts.getOrDefault(augmentedComment.getArticleId(), 0L) + 1);
+
+                assertThat(
+                    pass,
+                    report,
+                    "augmentedArticle missing",
+                    consumedAugmentedArticles,
+                    hasKey(augmentedComment.getArticleId())
+                );
+                final AugmentedArticle augmentedArticle = consumedAugmentedArticles.get(augmentedComment.getArticleId());
+                if (augmentedArticle != null) {
+                    assertThat(
+                        pass,
+                        report,
+                        "augmented articlePrefix mismatch",
+                        augmentedArticle.getText(),
+                        startsWith(augmentedComment.getArticlePrefix())
+                    );
+                }
+
+                assertThat(
+                    pass,
+                    report,
+                    "article missing",
+                    consumedArticles,
+                    hasKey(augmentedComment.getArticleId())
+                );
+                final Article article = consumedArticles.get(augmentedComment.getArticleId());
+                if (article != null) {
+                    assertThat(
+                        pass,
+                        report,
+                        "articlePrefix mismatch",
+                        article.getText(),
+                        startsWith(augmentedComment.getArticlePrefix())
+                    );
+                }
+            }
+
+
+            for (final RelationalSmokeTest.AugmentedArticle augmentedArticle : consumedAugmentedArticles.values()) {
+                assertThat(
+                    pass,
+                    report,
+                    "article " + augmentedArticle.getKey() + " comment count mismatch",
+                    augmentedArticle.getCommentCount(),
+                    is(commentCounts.getOrDefault(augmentedArticle.getKey(), 0L))
+                );
+            }
+
+            if (logResults) {
+                if (pass.get()) {
+                    LOG.info(
+                        "Evaluation passed ({}/{}) articles and ({}/{}) comments",
+                        consumedAugmentedArticles.size(),
+                        consumedArticles.size(),
+                        consumedAugmentedComments.size(),
+                        consumedComments.size()
+                    );
+                } else {
+                    LOG.error("Evaluation failed\n{}", report);
```

Comment:
When error, could we print all the received records from all four topics too? Just 2 cents from my past flaky test fighting experience.